Resume

DavidASC20 opened the pull request:
# Pull Request

## Description

Please include a summary of the changes and the related issue (if applicable). Please also include relevant motivation and context. List any dependencies that are required for this change (if applicable).

I added a dropdown list of gender, Veteran status, Ethnicity, and Disabled status, which are usually asked by companies in the recruiting process.

Fixes # (issue)
Fixes issue 16 with the addition of more settings that could be inputted by the user.
## Type of change

- [ ] Bug fix (change which fixes an issue)
- [ X] Enhancement (change which alters existing functionality)
- [ ] New feature (change which adds new functionality)
- [ ] This change requires a wiki/documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
I ran the code in the extension, verifying the choices.


## Checklist

- [X ] My code follows the style guidelines of this project
- [ X] I have performed a self-review of my code
- [ X] I have commented my code, particularly in hard-to-understand areas
- [ X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

Try to checkoff as much as possible if not everything!

Commit:
added Gender, disability status, ethnicity, and veteran status

Commit:
changed the text on them from the programming languages to the disability prompts

hli2238 committed:
PDF parser

Code to parse pdf into categories.

anthonyfabius commented on PDFParse/Hongwei_resume.pdf:
You probably want to remove this

anthonyfabius commented on PDFParse/output.txt:
The output file should be JSON so that is is compatible with chrome local storage. So instead of a output.txt we should have a output.json. Later on we will wont even want an output file and just have it write straight to local storage. One way to format this text as json would be 
```json
{
  "name": "HONGWEI LI (DILLON)",
  "phone": "(917) 702-5588",
  "email": "dillon.hongwei.li@gmail.com",
  "location": "New York, NY",
  "summary": "A highly motivated student seeking challenging summer software internship opportunities.",
  "education": [
    {
      "school": "Rensselaer Polytechnic Institute",
      "location": "Troy, NY",
      "degree": "B.S. in Computer Science",
      "gpa": "3.26/4.0",
      "graduation_date": "May 2026",
      "relevant_courses": [
        "Data Structures",
        "Computer Science I",
        "Business and Management"
      ]
    },
    {
      "school": "Stuyvesant High School",
      "location": "New York, NY",
      "degree": "High School Diploma",
      "gpa": "3.5/4.00",
      "graduation_date": "Jun 2022",
      "relevant_courses": [
        "Computer Science",
        "Technical Drawing",
        "3D Modeling",
        "Analog and Applied Electronics"
      ]
    }
  ],
  "professional_experience": [
    {
      "position": "CodePath Android Development",
      "location": "New York, NY",
      "dates": "Feb 2023 – Apr 2023",
      "description": [
        "Learn the basics of Android development including IDEs, Kotlin language, and debugging.",
        "Design and architect a functional multi-screen mobile app in a group setting."
      ]
    },
    {
      "position": "StuyPulse FIRST Robotics Team",
      "location": "New York, NY",
      "dates": "Sep 2019 – Jun 2022",
      "description": [
        "Designed and built prototypes of a feeding mechanism that allows the robot to collect and store balls for shooting.",
        "Gained hands-on experience in machines such as drill press, laser cutter, bandsaw, and belt sander."
      ]
    },
    {
      "position": "Technical Project",
      "location": "New York, NY",
      "dates": "Sep 2021 – Jun 2022",
      "description": [
        "Designed and built circuitry to adjust the LED and seven-segment display using IC chips, ceramic capacitors, resistors, potentiometers, and buttons.",
        "Utilized AutoCAD to design a 3D water bottle jug and created a multi-view drawing with dimensions.",
        "Laser cut a wooden, finger-jointed light box with engraved images and cut-outs."
      ]
    },
    {
      "position": "SAT Prep Center",
      "location": "New York, NY",
      "dates": "Jul 2020 – Aug 2021",
      "description": [
        "Tutored multiple groups of 15 students to enhance their mathematical and English comprehension skills."
      ]
    }
  ],
  "leadership_development": [
    {
      "organization": "Society of Asian Scientists and Engineers (SASE)",
      "location": "Troy, NY",
      "role": "Member, Mentee",
      "dates": "Sep 2022 – Present",
      "description": [
        "Developed professional experience and network through participating in various conferences, workshops, and social events.",
        "Aiming to join the Public Relations Committee to raise awareness of social and service events to the community."
      ]
    },
    {
      "organization": "Eighth Wonder Dance Club",
      "location": "Troy, NY",
      "role": "Treasurer",
      "dates": "Sep 2022 – Present",
      "description": [
        "Handled all income and expenditure of the club while tracking the club funds in a financial report to organize and plan future club events.",
        "Will be coordinating with school officials and monitoring events to request and raise funds for the club."
      ]
    },
    {
      "organization": "PSAL Stunt & Cheer",
      "location": "New York, NY",
      "role": "Cheerleader and Back spotter",
      "dates": "Sep 2018 – Jun 2022",
      "description": [
        "Participated in 5 cheerleading events to support the school’s sports games.",
        "Supervised and mentored new stunt members to facilitate their transition to the team.",
        "Successfully organized 3 in-school fundraisers for the team to participate in national competitions."
      ]
    }
  ],
  "skills": {
    "technical": ["Python", "HTML", "Kotlin", "Scratch", "AutoCAD"],
    "prototyping": ["3D printer", "laser cutter", "horizontal bandsaw", "drill press"],
    "languages": ["Proficient in English and Mandarin"]
  }
}
```

Essentially we want to break it down into pieces so that when the AI model attempts to autofill something like work experience, it can take the json sections called "Professional Experience" and easily see what goes where as we have sections for the position, title, dates, description, etc. Also txt files aren't compatible with local storage and we definitely don't want to parse a resume every time we want to fill in data.

anthonyfabius commented on PDFParse/package.json:
Why is there another package.json? Shouldn't you add these dependencies to the package.json in the working directory.

anthonyfabius commented on PDFParse/pdfparse.js:
Git ignore this

anthonyfabius commented on PDFParse/pdfsort.js:
Git ignore this

anthonyfabius commented on RPIVerificationWorksheet_662048737_Hongwei.Li.pdf:
Probably want to remove this as well.

anthonyfabius commented on parsePdf.js:
Git ignore this

anthonyfabius commented on PDFParse/pdfparse.ts:
```diff
@@ -0,0 +1,39 @@
+
+async function convertPdfToTxt(): Promise<void> {
+  try {
+    // Get PDF file path from user input
+    const pdfPath = readlineSync.question('Enter the path to the PDF file: ');
+
+    // Check if the file exists
+    if (!fs.existsSync(pdfPath)) {
+      console.error('Error: The specified file does not exist.');
+      return;
+    }
```
Getting files is gonna work a little differently that this but for testing this is file I guess. Rather then having the user enter the path we should just have a set path like /PDFParse/parse.pdf would be the default name.

anthonyfabius commented on PDFParse/pdfparse.ts:
```diff
@@ -0,0 +1,39 @@
+    const textContent = data.text;
+
+    // Create output file path
+    const txtPath = path.join(__dirname, 'output.txt');
```
Output as json

anthonyfabius commented on react-chrome-extension/src/options/App.tsx:
This entire modification is in the wrong branch. You probably want to include this in the SettingsPage branch and not this one.

anthonyfabius commented on parsePdf.ts:
parsePdf and pdfparse... probably some better ways to name these 2 files. As for content I am loosely understanding the workflow of this function and the ones it calls. Adding a single line comment to the top of all the pdf parsing files that explain what each do would be helpful.

anthonyfabius commented on PDFParse/pdfsort.ts:
```diff
@@ -0,0 +1,62 @@
+  content: string[];
+}
+
+function isName(line: string, isFirstNonEmptyLine: boolean): string | null {
+  const trimmedLine = line.trim();
+  return isFirstNonEmptyLine && trimmedLine.length > 0 ? trimmedLine : null;
+}
```
What is the point of this function?

anthonyfabius commented on PDFParse/pdfsort.ts:
```diff
@@ -0,0 +1,62 @@
+      return;
+    }
+
+    // Check if the line is the name
+    const name = isName(line, isFirstNonEmptyLine);
+    if (name) {
+      parsedInfo.push({ category: 'Name', content: [name] });
+      isFirstNonEmptyLine = false; // Set to false after finding the first non-empty line
+      return;
+    }
```
What is "name" supposed to be. At first glance I keep thinking this is the name of the person but then I think its  a position name or a section heading.

anthonyfabius reviewed (changes requested):
Reviewed each files.  The main things I noticed that need to be fixed:
 - Some random artifacts are still present that we don't want
 - Making the parsing output a json file and not a txt
 - Removing the settings page changes
 - Naming conventions of various files and variables
 - Sparce comments in hard to understand areas

hli2238 committed:
New output destination

Changed the output to be inside a json file instead of a txt file

mik0lam committed:
Test cases

added some test cases, and made the final output into a json file

anthonyfabius committed:
Updating dependencies and git ignoring artifacts

Got rid of the npm dependencies nested in PDF Parse. Also uncached all pdf parse files and git ignored the ones we don't care about.

hli2238 committed:
Duplicate files

Removed duplicate/useless files that are in wrong folders.

anthonyfabius committed:
Added back JS files

my bad :skull:

mik0lam committed:
created instructions

instructions for ease of use of the program(s)

mik0lam committed:
pdf parse change

changed how the pdf parser works, by changing the sorting through keywords.

mik0lam committed:
changed keywords

added more keywords to help parse

MBtheOtaku committed:
added location, email addresses, and phone number for contact information

mik0lam committed:
cutting down

cut down on the amount of files by combining pdf parse and pdf sort

hli2238 committed:
Simple version of code to parse cover letters.

The code currently parses the string of the cover letter into the name and body only.

Co-Authored-By: mik0lam <144414594+mik0lam@users.noreply.github.com>
Co-Authored-By: MBtheOtaku <45703141+MBtheOtaku@users.noreply.github.com>

Commit:
adding cover letter for testing

mik0lam committed:
Instructions

updated instructions to reflect new code

mik0lam committed:
Cover Letter

Changed up the cover letter parse, is now a stable prototype, has name body and address sections.

Co-Authored-By: MBtheOtaku <45703141+MBtheOtaku@users.noreply.github.com>
Co-Authored-By: hli2238 <121896329+hli2238@users.noreply.github.com>

mik0lam committed:
removed example

got rid of unnecessary files and lines of code

mik0lam committed:
Implemented db code

Added a prototype to pull the resume from the database

hli2238 committed:
Database in cover letter parser

Modified the code to also implement the db function in the cover letter parser so the code can take in a file from the database instead of a manual inputted string

Co-Authored-By: mik0lam <144414594+mik0lam@users.noreply.github.com>
Co-Authored-By: MBtheOtaku <45703141+MBtheOtaku@users.noreply.github.com>

hli2238 committed:
Saving parsed data

Added a function to store the parsed info into the database.

Co-Authored-By: mik0lam <144414594+mik0lam@users.noreply.github.com>
Co-Authored-By: MBtheOtaku <45703141+MBtheOtaku@users.noreply.github.com>

hli2238 committed:
Adding the save implementation to pdfParse

This function saves the parsed data into the database for future use just like in the cover letter parser.

hli2238 committed:
Comments

Added comments to the code.
